buildSrc/subprojects/buildquality/src/main/kotlin/org/gradle/gradlebuild/buildquality/incubation/IncubatingApiReportTask.kt

```diff
@@ -0,0 +1,274 @@
+package org.gradle.gradlebuild.buildquality.incubation
+
+import com.github.javaparser.JavaParser
+import com.github.javaparser.ast.CompilationUnit
+import com.github.javaparser.ast.Node
+import com.github.javaparser.ast.body.AnnotationDeclaration
+import com.github.javaparser.ast.body.ClassOrInterfaceDeclaration
+import com.github.javaparser.ast.body.EnumDeclaration
+import com.github.javaparser.ast.body.MethodDeclaration
+import com.github.javaparser.ast.nodeTypes.NodeWithAnnotations
+import com.github.javaparser.ast.nodeTypes.NodeWithJavadoc
+import com.github.javaparser.ast.nodeTypes.NodeWithSimpleName
+import com.github.javaparser.javadoc.description.JavadocSnippet
+import com.github.javaparser.symbolsolver.JavaSymbolSolver
+import com.github.javaparser.symbolsolver.resolution.typesolvers.CombinedTypeSolver
+import com.github.javaparser.symbolsolver.resolution.typesolvers.JavaParserTypeSolver
+import com.github.javaparser.symbolsolver.resolution.typesolvers.ReflectionTypeSolver
+import org.gradle.api.DefaultTask
+import org.gradle.api.file.FileCollection
+import org.gradle.api.file.RegularFileProperty
+import org.gradle.api.plugins.JavaPluginConvention
+import org.gradle.api.provider.Property
+import org.gradle.api.tasks.CacheableTask
+import org.gradle.api.tasks.Input
+import org.gradle.api.tasks.InputFile
+import org.gradle.api.tasks.InputFiles
+import org.gradle.api.tasks.OutputFile
+import org.gradle.api.tasks.TaskAction
+import org.gradle.api.tasks.PathSensitive
+import org.gradle.api.tasks.PathSensitivity
+import org.gradle.workers.IsolationMode
+import org.gradle.workers.WorkerExecutor
+import java.io.BufferedReader
+import java.io.File
+import java.util.concurrent.TimeUnit
+import javax.inject.Inject
+
+
+@CacheableTask
+open class IncubatingApiReportTask @Inject constructor(private val workerExecutor: WorkerExecutor) : DefaultTask() {
+
+    @InputFile
+    @PathSensitive(PathSensitivity.RELATIVE)
+    val versionFile: RegularFileProperty = project.objects.fileProperty()
+
+    @InputFile
+    @PathSensitive(PathSensitivity.RELATIVE)
+    val releasedVersionsFile: RegularFileProperty = project.objects.fileProperty()
+
+    @Input
+    val title: Property<String> = project.objects.property(String::class.java).also {
+        it.set(project.provider { project.name })
+    }
+
+    @InputFiles
+    @PathSensitive(PathSensitivity.RELATIVE)
+    val sources: Property<FileCollection> = project.objects.property(FileCollection::class.java).also {
+        it.set(project.provider {
+            project.convention
+                .getPlugin(JavaPluginConvention::class.java)
+                .sourceSets
+                .getByName("main")
+                .java
+                .sourceDirectories
+        })
+    }
```

Comment:
Why not use `SourceTask` instead and just visit the `source: FileTree`? The defaults don't seem to be used at all.

Reply:
I just copied from my external plugin version. We don't necessarily need defaults here, but I wanted reasonable defaults.